internal/pkg/build/assemblers/sif.go

```diff
@@ -189,38 +190,59 @@ func (a *SIFAssembler) Assemble(b *types.Bundle, path string) error {
 	}
 	sylog.Verbosef("Set SIF container architecture to %s", arch)
 
-	if err := s.Create([]string{b.RootfsPath}, fsPath, flags); err != nil {
-		return fmt.Errorf("while creating squashfs: %v", err)
-	}
-
 	var encOpts *encryptionOptions
-
-	if b.Opts.EncryptionKeyInfo != nil {
-		plaintext, err := cryptkey.NewPlaintextKey(*b.Opts.EncryptionKeyInfo)
-		if err != nil {
-			return fmt.Errorf("unable to obtain encryption key: %+v", err)
+	if b.Opts.Unprivilege {
+		sylog.Debugf("Creating squashfs image and will use gocryptfs")
+		if b.Opts.EncryptionKeyInfo == nil {
+			return fmt.Errorf("unprivileged encryption build requires valid encryption key or passphrase to work properly")
```

Comment:
` Surprisingly, privileged encryption without any environment variable or passphrase succeeds, although I don't know if it could ever be unencrypted.`
how did you test this?
I've tried different combinations
```
vagrant@ubuntu:~/test$ apptainer build --encrypt --force 1.sif busybox_arm64.sif 
INFO:    Starting build...
INFO:    Verifying bootstrap image busybox_arm64.sif
WARNING: failed to get key material: 404 Not Found
WARNING: integrity: signature object 5 not valid: openpgp: signature made by unknown entity
WARNING: Bootstrap image could not be verified, but build will continue.
INFO:    Creating SIF file...
ERROR:   no encryption key environment variable or --passphrase provided
FATAL:   While performing build: no encryption key environment variable or --passphrase provided
```

```
vagrant@ubuntu:~/test$ apptainer build --pem-path 1.sif busybox_arm64.sif 
Error for command "build": accepts 2 arg(s), received 1

Usage:
  apptainer [global options...] build [local options...] <IMAGE PATH> <BUILD SPEC>

Run 'apptainer --help' for more detailed usage information.
```

```
vagrant@ubuntu:~/test$ apptainer build --passphrase 1.sif busybox_arm64.sif 
Build target '1.sif' already exists and will be deleted during the build process. Do you want to continue? [N/y] y
Enter encryption passphrase: 
FATAL:   Cannot encrypt container with empty passphrase
```

Reply:
I've added sylog error in this commit
https://github.com/apptainer/apptainer/pull/1185/commits/523354c8a4daec2223e38bc1de95114df7f38ef3

Reply:
Oh, no, a sylog error here does not provide an earlier notification, it just ends up giving two duplicate error messages in a row:
```
ERROR:   no encryption key environment variable or --passphrase provided
FATAL:   While performing build: no encryption key environment variable or --passphrase provided
```
So please remove that extra sylog.Errorf().  I was talking about adding a check much earlier, at the place where it would normally prompt for the passphrase, so it would die before doing any building.